lib/ansible/modules/web_infrastructure/ansible_tower/tower_organization.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python
+#coding: utf-8 -*-
+
+# (c) 2016, Wayne Witzel III <wayne@riotousliving.com>
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: tower_organization
+version_added: "2.3"
+short_description: create, update, or destroy Ansible Tower organizations
+description:
+    - Create, update, or destroy Ansible Tower organizations. See
+      U(https://www.ansible.com/tower) for an overview.
+options:
+    name:
+      description:
+        - Name string to use for the organization.
+      required: True
+      default: null
+    description:
+      description:
+        - String to use for the organization.
+      required: False
+      default: null
+    state:
+      description:
+        - Desired state of the resource.
+      required: False
+      default: "present"
+      choices: ["present", "absent"]
+    config_file:
+      description:
+        - Path to the Tower config file. See notes.
+      required: False
+      default: null
+
+
+requirements:
+  - "python >= 2.6"
+  - "ansible-tower-cli >= 3.0.3"
+
+notes:
+  - If no I(config_file) is provided we will attempt to use the tower-cli library
+    defaults to find your Tower host information.
+  - I(config_file) should contain Tower configuration in the following format:
+      host=hostname
+      username=username
+      password=password
+'''
+
+
+EXAMPLES = '''
+  - tasks
+    - name: Create organization
+      tower_organization:
+          name: "Foo"
+          description: "Foo bar organization"
+          state: present
+          config_file: "~/tower_cli.cfg"
+'''
+
+import os
+
+try:
+    import tower_cli
+    import tower_cli.utils.exceptions as exc
+    from tower_cli.utils import parser
+    from tower_cli.conf import settings
+
+    HAS_TOWER_CLI = True
+except ImportError:
+    HAS_TOWER_CLI = False
+
+
+def tower_auth_config(module):
```

Comment:
i would consider moving this to a module_utils/tower as I expect other modules will need to use the same auth